vendor/bat-native-ads/src/bat/ads/internal/eligible_ads/sample_ads_unittest.cc

```diff
@@ -0,0 +1,153 @@
+
+  AdPredictorInfo<CreativeAdNotificationInfo> ad_predictor_3;
+  ad_predictor_3.score = 0;
+  ads[base::GenerateGUID()] = ad_predictor_3;
+
+  // Act
+  const absl::optional<CreativeAdNotificationInfo> ad = SampleFromAds(ads);
+
+  // Assert
+  const absl::optional<CreativeAdNotificationInfo> expected_ad = absl::nullopt;
+  EXPECT_EQ(expected_ad, ad);
+}
+
+TEST(BatAdsSampleAdsTest, DeterministicallySampleFromAdsWithOneNonZeroScore) {
+  // Arrange
+  CreativeAdNotificationPredictorMap ads;
+
+  CreativeAdNotificationInfo creative_ad_notification_1 =
+      GetCreativeAdNotification("foo-bar", 1.0, 1);
+  AdPredictorInfo<CreativeAdNotificationInfo> ad_predictor_1;
+  ad_predictor_1.score = 0;
+  ad_predictor_1.creative_ad = creative_ad_notification_1;
+  ads[creative_ad_notification_1.creative_instance_id] = ad_predictor_1;
+
+  CreativeAdNotificationInfo creative_ad_notification_2 =
+      GetCreativeAdNotification("foo-bar", 1.0, 1);
+  AdPredictorInfo<CreativeAdNotificationInfo> ad_predictor_2;
+  ad_predictor_2.score = 0.1;
+  ad_predictor_2.creative_ad = creative_ad_notification_2;
+  ads[creative_ad_notification_2.creative_instance_id] = ad_predictor_2;
+
+  CreativeAdNotificationInfo creative_ad_notification_3 =
+      GetCreativeAdNotification("foo-bar", 1.0, 1);
+  AdPredictorInfo<CreativeAdNotificationInfo> ad_predictor_3;
+  ad_predictor_3.score = 0;
+  ad_predictor_3.creative_ad = creative_ad_notification_3;
+  ads[creative_ad_notification_3.creative_instance_id] = ad_predictor_3;
+
+  // Act
+  for (int i = 0; i < 10; i++) {
+    CreativeAdNotificationInfo expected_ad = creative_ad_notification_2;
+    const absl::optional<CreativeAdNotificationInfo> ad = SampleFromAds(ads);
+    EXPECT_EQ(expected_ad, ad.value());
+  }
+
+  // Assert
+}
+
+TEST(BatAdsSampleAdsTest, ProbabilisticallySampleFromAds) {
+  // Arrange
+  CreativeAdNotificationPredictorMap ads;
+
+  CreativeAdNotificationInfo creative_ad_notification_1 =
+      GetCreativeAdNotification("foo-bar", 1.0, 1);
+  AdPredictorInfo<CreativeAdNotificationInfo> ad_predictor_1;
+  ad_predictor_1.creative_ad = creative_ad_notification_1;
+  ad_predictor_1.score = 3;
+  ads[creative_ad_notification_1.creative_instance_id] = ad_predictor_1;
+
+  CreativeAdNotificationInfo creative_ad_notification_2 =
+      GetCreativeAdNotification("foo-bar", 1.0, 1);
+  AdPredictorInfo<CreativeAdNotificationInfo> ad_predictor_2;
+  ad_predictor_2.creative_ad = creative_ad_notification_2;
+  ad_predictor_2.score = 3;
+  ads[creative_ad_notification_2.creative_instance_id] = ad_predictor_2;
+
+  // Act
+  int ads_1_count = 0;
+  int ads_2_count = 0;
+
+  // P(X>1) > 0.99999999 with X~Bin(n=25, p=0.5), i.e. less than 1 in 100M tests
+  // are expected to fail
+  for (int i = 0; i < 25; i++) {
+    const absl::optional<CreativeAdNotificationInfo> ad = SampleFromAds(ads);
+
+    if (ad.value().creative_instance_id ==
+        creative_ad_notification_1.creative_instance_id) {
+      ads_1_count += 1;
```

Comment:
`++;` instead of `+= 1;`. Worth globally searching for this

Reply:
fixed